pandas/core/indexes/base.py

```diff
@@ -1959,537 +2416,432 @@ def get_duplicates(self):
-
-    def _validate_index_level(self, level):
-        """
-        Validate index level.
-
-        For single-level Index getting level number is a no-op, but some
-        verification must be done like in MultiIndex.
-
+    def _get_unique_index(self, dropna=False):
         """
-        if isinstance(level, int):
-            if level < 0 and level != -1:
-                raise IndexError("Too many levels: Index has only 1 level,"
-                                 " %d is not a valid level number" % (level, ))
-            elif level > 0:
-                raise IndexError("Too many levels:"
-                                 " Index has only 1 level, not %d" %
-                                 (level + 1))
-        elif level != self.name:
-            raise KeyError('Level %s must be same as name (%s)' %
-                           (level, self.name))
-
-    def _get_level_number(self, level):
-        self._validate_index_level(level)
-        return 0
+        Returns an index containing unique values.
 
-    @cache_readonly
-    def inferred_type(self):
-        """
-        Return a string of the type inferred from the values.
-        """
-        return lib.infer_dtype(self)
+        Parameters
+        ----------
+        dropna : bool
+            If True, NaN values are dropped.
 
-    def _is_memory_usage_qualified(self):
-        """
-        Return a boolean if we need a qualified .info display.
+        Returns
+        -------
+        uniques : index
         """
-        return self.is_object()
-
-    def is_type_compatible(self, kind):
-        return kind == self.inferred_type
-
-    @cache_readonly
-    def is_all_dates(self):
-        if self._data is None:
-            return False
-        return is_datetime_array(ensure_object(self.values))
+        if self.is_unique and not dropna:
+            return self
 
-    def __reduce__(self):
-        d = dict(data=self._data)
-        d.update(self._get_attributes_dict())
-        return _new_Index, (self.__class__, d), None
+        values = self.values
 
-    def __setstate__(self, state):
-        """
-        Necessary for making this object picklable.
-        """
+        if not self.is_unique:
+            values = self.unique()
 
-        if isinstance(state, dict):
-            self._data = state.pop('data')
-            for k, v in compat.iteritems(state):
-                setattr(self, k, v)
+        if dropna:
+            try:
+                if self.hasnans:
+                    values = values[~isna(values)]
+            except NotImplementedError:
+                pass
 
-        elif isinstance(state, tuple):
+        return self._shallow_copy(values)
 
-            if len(state) == 2:
-                nd_state, own_state = state
-                data = np.empty(nd_state[1], dtype=nd_state[2])
-                np.ndarray.__setstate__(data, nd_state)
-                self.name = own_state[0]
```

Comment:
indexing methods

Reply:
what method was this comment supposed to refer to?

Reply:
oh this section, basically group all of the indexing (and indexer conversion methods)

Reply:
IOW also include ``__getitem__``

Reply:
Not asking what section, asking what method.  On GH your comment is pointing towards `__setstate__`, which I really doubt you mean should go in the indexing section